letting_context.md

```diff
@@ -175,6 +181,155 @@ A user might change the reference of a unit in GARAIO REM which affects tenancy
+
+#### Example
+
+```json
+{
+  "eventType":"Letting.TenancyAgreement.DraftActivated",
+  "data": {
+    "reference":"1234.01.0001.01",
+    "startDate":"2025-08-03",
+    "state":"activated"
+  }
+}
+```
+
+### Letting.TenancyAgreement.Updated
+
+The activated tenancy agreement has some new updates. 
+
+| Field                    | Type     | Content / Remarks                                                                                                                     |
+|--------------------------| -------- |---------------------------------------------------------------------------------------------------------------------------------------|
+| eventType                | `string` | Letting.TenancyAgreement.Updated                                                                                                 |
+| data                     | `hash`   |                                                                                                                                       |
+| &nbsp;&nbsp;reference    | `string` | unique tenancy agreement identifier, eg '1234.01.0001.01'                                                                             |
+| &nbsp;&nbsp;startDate    | `string` | The starting date of the tenancy agreement. |
+| &nbsp;&nbsp;state        | `string` | The current state of the tenancy agreement. It can be in validated, activated, deactivated, cancelled state                           |
+
+#### Example
+
+```json
+{
+  "eventType":"Letting.TenancyAgreement.Updated",
+  "data": {
+    "reference":"1234.01.0001.01",
+    "startDate":"2025-08-03",
+    "state":"activated"
+  }
+}
+```
+
+### Letting.TenancyAgreement.Terminated
+
+The tenancy agreement has been cancelled.
+
+| Field                    | Type     | Content / Remarks                                                                                                                     |
+|--------------------------| -------- |---------------------------------------------------------------------------------------------------------------------------------------|
+| eventType                | `string` | Letting.TenancyAgreement.Terminated                                                                                                 |
+| data                     | `hash`   |                                                                                                                                       |
+| &nbsp;&nbsp;reference    | `string` | unique tenancy agreement identifier, eg '1234.01.0001.01'                                                                             |
+| &nbsp;&nbsp;state        | `string` | The current state of the tenancy agreement. It can be in validated, activated, deactivated, cancelled state                           |
+
+#### Example
+
+```json
+{
+  "eventType":"Letting.TenancyAgreement.Terminated",
+  "data": {
+    "reference":"1234.01.0001.01",
```

Comment:
All messages should include a startDate to allow the consumer to identify the correct version of the agreement. Please make sure we send & test it for all newly implemented messages and update the docs accordingly.